src/lib/doctrine-dbal-bulk/src/Flow/Doctrine/Bulk/BulkData.php

```diff
@@ -143,6 +144,16 @@ public function toSqlParameters(TableDefinition $table) : array
                             Types::DATETIME_IMMUTABLE => \DateTimeImmutable::createFromMutable($entry),
                             default => $entry,
                         },
+                        \DOMDocument::class => match (Type::getTypeRegistry()->lookupName($table->dbalColumn($column)->getType())) {
+                            Types::TEXT,
+                            Types::STRING => $entry->saveXML($entry->documentElement),
+                            default => $entry,
+                        },
+                        \DOMElement::class => match (Type::getTypeRegistry()->lookupName($table->dbalColumn($column)->getType())) {
```

Comment:
@jmortlock I followed your suggestion, this requires a bigger refactoring, it's not perfect but it works, and it works well so I'm keeping it for now. Ideally I would like to extract that logic to a different place but that requires some planning and I'm a bit short on time currently

Reply:
Works for me